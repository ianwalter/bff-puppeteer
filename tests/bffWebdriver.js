const { test, run } = require('@ianwalter/bff')

const toName = ({ name, err }) => name + (err ? `: ${err}` : '')

test('bff-webdriver', async ({ expect }) => {
  const result = await run({ plugins: ['.'] })
  expect(result.filesRegistered).toBe(3)
  expect(result.testsRegistered).toBe(6)
  expect(result.testsRun).toBe(7)
  expect(result.passed.length).toBe(3)
  expect(result.passed.map(toName).sort()).toMatchSnapshot()
  expect(result.failed.length).toBe(2)
  expect(result.failed.map(toName).sort()).toMatchSnapshot()
  expect(result.skipped.length).toBe(2)
  expect(result.skipped.map(toName).sort()).toMatchSnapshot()
})
