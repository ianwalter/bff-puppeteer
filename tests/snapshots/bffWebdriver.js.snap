// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`bff-webdriver 1`] = `
Array [
  "arithmetic",
  "dom snapshot",
  "manual pass",
]
`;

exports[`bff-webdriver 2`] = `
Array [
  "deep equal failure: expect(received).toBe(expected) // Object.is equality

If it should pass with deep equality, replace \\"toBe\\" with \\"toStrictEqual\\"

Expected: {\\"name\\": \\"Joe\\"}
Received: serializes to the same string",
  "property of undefined: Cannot read property 'that' of undefined",
]
`;

exports[`bff-webdriver 3`] = `
Array [
  "skip no assertions",
  "skip with no test function",
  "skip with test specified after name",
]
`;
